src/block/tile/Furnace.php

```diff
@@ -212,7 +212,13 @@ public function onUpdate() : bool{
 			}
 			$ret = true;
 			if(mt_rand(1, 100) < 5){
-				$this->fireCrackle();
+				if($this instanceof BlastFurnace){
```

Comment:
If you ever find yourself doing `this instanceof`, you're doing something crazy that defeats the point of OOP.

also, why is this in the tile...

Reply:
Indeed ...
However, the fact that this is in the tile is that all the behavior of the current furnaces are in it. Should all instances of furnace be required to specify the sound they play when in action ?

Reply:
Putting it in the tile just means it will have to be moved again later, which wastes everyone's time. It should be done in Furnace->onScheduledUpdate() instead.

Also, you should be able to use FurnaceType instead of looking at the tile class.